app/src/main/java/co/condorlabs/customcomponents/customcollapsibleview/CollapsibleView.kt

```diff
@@ -0,0 +1,555 @@
+                            if (from > to) from else to
+                        )
+                    )
+                }
+            } ?: return
+
+        startValueAnimator(animator)
+    }
+
+    private fun startValueAnimator(valueAnimator: ValueAnimator) {
+        AnimatorSet().apply {
+            play(valueAnimator)
+            interpolator = AccelerateDecelerateInterpolator()
+            start()
+        }
+    }
+
+    private fun updateContentHeight(height: Int = ZERO) {
+        flContent.apply {
+            layoutParams?.height = height
+            requestLayout()
+        }
+    }
+
+    private fun rotateIndicatorArrow(rotation: Float = ZERO_FLOAT) {
+        ivActionIndicator.rotation = -rotation
+    }
+
+    private fun rotationIndicatorArrowValue(currentValue: Int, limit: Int): Float {
+        return (COLLAPSIBLE_INDICATOR_ROTATION * ((currentValue * ONE_HUNDRED_FLOAT) / limit)) / ONE_HUNDRED_FLOAT
+    }
+
+    private fun updateActionLabel(isCollapsed: Boolean) {
+        tvActionLabel.text = if (isCollapsed) {
+            showActionLabel
+        } else {
+            hideActionLabel
+        }
+    }
+
+    private fun callOnCollapse(isCollapsed: Boolean): Boolean {
+        return onCollapseListener?.let { it.onCollapse(isCollapsed); true } ?: false
+    }
+
+    private fun Int.getMinimumCollapsibleContentPadding(): Int {
+        return if (this < MIN_DEFAULT_PADDING) MIN_DEFAULT_PADDING else this
+    }
+
+    fun setImage(imageResourceId: Int) {
+        if (imageResourceId != NO_ID) {
+            val imageDrawableResource = AppCompatResources.getDrawable(context, imageResourceId)
+            ivCollapsible.setImageDrawable(imageDrawableResource)
+            ivCollapsible.visibility = View.VISIBLE
+        } else {
+            ivCollapsible.visibility = View.GONE
+        }
+    }
+
+    fun setImageTint(colorTint: Int) {
+        if (colorTint != NO_ID) {
+            ImageViewCompat.setImageTintList(
+                ivCollapsible,
+                AppCompatResources.getColorStateList(context, colorTint)
+            )
+        }
+    }
+
+    fun setContent(collapsibleContent: View) {
+        flContent.addView(collapsibleContent)
+
+        val displayMetrics = DisplayMetrics()
+        (context as? AppCompatActivity)?.windowManager?.defaultDisplay?.getMetrics(displayMetrics)
+            ?.let {
+                flContent.measure(
+                    displayMetrics.widthPixels,
+                    displayMetrics.heightPixels
+                )
+                contentHeight = flContent.measuredHeight
+            }
+
+        if (isCollapsedContent) {
+            updateContentHeight()
+            rotateIndicatorArrow()
+            callOnCollapse(isCollapsedContent)
+        }
+
+        updateActionLabel(isCollapsedContent)
+
+        setOnClickListener { collapseContent(contentHeight) }
+    }
+
+    fun visibleIndicatorArrow(isVisible: Boolean) {
```

Comment:
what is thing doing visibleIndicatorArrow

Reply:
changed